src/app/domain/backup/services/manageRemoteBackupConfig.ts

```diff
@@ -0,0 +1,148 @@
+    }
+  }
+}
+
+const isInTrash = (path: string): boolean => {
+  return path.startsWith('/.cozy_trash')
+}
+
+export const fetchRemoteBackupConfigs = async (
+  client: CozyClient
+): Promise<RemoteBackupConfig[]> => {
+  const { included } = await client.collection(DOCTYPE_FILES).findReferencedBy({
+    _type: DOCTYPE_APPS,
+    _id: BACKUP_REF
+  })
+
+  const backupFolders = included as BackupFolder[]
+
+  const remoteBackupConfigs: RemoteBackupConfig[] = backupFolders
+    .filter(folder => !isInTrash(folder.attributes.path))
+    .map(backupFolder => ({
+      backupFolder: {
+        id: backupFolder._id,
+        name: backupFolder.attributes.name,
+        path: backupFolder.attributes.path
+      },
+      backupDeviceIds: backupFolder.attributes.metadata?.backupDeviceIds ?? []
+    }))
+
+  return remoteBackupConfigs
+}
+
+export const isRemoteBackupConfigFromDevice = (
+  remoteBackupConfig: RemoteBackupConfig,
+  deviceId: string
+): boolean => {
+  return remoteBackupConfig.backupDeviceIds.includes(deviceId)
+}
+
+export const fetchDeviceRemoteBackupConfig = async (
+  client: CozyClient
+): Promise<RemoteBackupConfig | undefined> => {
+  const remoteBackupConfigs = await fetchRemoteBackupConfigs(client)
+  const deviceId = await getDeviceId()
+
+  const remoteBackupConfig = remoteBackupConfigs.find(remoteBackupConfig =>
+    isRemoteBackupConfigFromDevice(remoteBackupConfig, deviceId)
+  )
+
+  return remoteBackupConfig
+}
+
+export const createRemoteBackupFolder = async (
+  client: CozyClient
+): Promise<RemoteBackupConfig> => {
+  const deviceName = await getDeviceName()
+  const deviceId = await getDeviceId()
+
+  const {
+    data: { _id: backupRootId }
+  } = await client
+    .collection(DOCTYPE_FILES)
+    .createDirectoryByPath(BACKUP_ROOT_PATH)
+
+  const backupFolderAttributes = {
+    name: deviceName,
+    dirId: backupRootId,
+    metadata: { backupDeviceIds: [deviceId] }
+  }
+
+  const {
+    data: { _id: backupFolderId }
+  } = await client
+    .collection(DOCTYPE_FILES)
+    .createDirectory(backupFolderAttributes)
+
+  await client.collection(DOCTYPE_FILES).addReferencesTo(
+    {
+      _id: BACKUP_REF,
+      _type: DOCTYPE_APPS
+    },
+    [
+      {
+        _id: backupFolderId
+      }
+    ]
+  )
+
+  const { data: backupFolder } = await client
+    .collection(DOCTYPE_FILES)
+    .get(backupFolderId)
+
+  const remoteBackupConfig = {
+    backupFolder: {
+      id: backupFolder._id,
+      name: backupFolder.name,
+      path: backupFolder.path
+    },
+    backupDeviceIds: backupFolder.metadata?.backupDeviceIds ?? []
+  }
+
+  return remoteBackupConfig
+}
+
+export const fetchBackupedMedias = async (
+  client: CozyClient,
+  deviceRemoteBackupConfig: RemoteBackupConfig
+): Promise<BackupedMedia[]> => {
+  const { backupFolder } = deviceRemoteBackupConfig
+
+  const filesQuery = buildFilesQuery(backupFolder.id)
+
+  const { data } = (await client.query(filesQuery)) as FilesQueryResult
```

Comment:
We'll only have the first 100 items with this query. If you really want all the files, you should use queryAll() or fetchMore()

Reply:
also, if you need only the filename, you should `select(['name'])`